packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -283,11 +271,9 @@ export class ZoweVsCodeExtension {
             await (zeRegister?.getCommonApi(serviceProfile).logout ?? Logout.apimlLogout)(updSession);
 
             const connOk = serviceProfile.profile.host === baseProfile.profile.host && serviceProfile.profile.port === baseProfile.profile.port;
-            if (connOk) {
-                await cache.updateBaseProfileFileLogout(baseProfile);
-            } else {
-                await cache.updateBaseProfileFileLogout(serviceProfile);
-            }
+            await cache.updateBaseProfileFileLogout(connOk ? baseProfile : serviceProfile);
+            serviceProfile.profile = { ...serviceProfile.profile, tokenType: undefined, tokenValue: undefined };
```

Comment:
I noticed `serviceProfile.profile` is reassigned as a new object - since this is in v2, I don't think this has any major side effects. However, it does make me wonder if we have similar logic in v3 and whether that logic works well with other parts of Zowe Explorer that keep profile references (such as tree nodes, filesystem) 🤔 

No request for changes here, it's merely an observation 😋